src/env/mod.rs

```diff
@@ -1,37 +1,24 @@
 mod as_env;
 
 use crate::as_execution::{abi_bail, ABIResult};
-use crate::{GasCosts, Interface};
+use crate::GasCosts;
 use wasmer::{AsStoreMut, Global};
 
 pub(crate) use as_env::*;
 
-macro_rules! get_memory {
-    ($env:ident) => {
-        match $env.get_wasm_env().memory.as_ref() {
-            Some(mem) => mem,
-            _ => abi_bail!("No memory in env"),
-        }
-    };
-}
-pub(crate) use get_memory;
-
-pub(crate) trait MassaEnv<T> {
-    fn new(interface: &dyn Interface, gas_costs: GasCosts) -> Self;
+/// Trait describing a metered object
+pub(crate) trait Metered {
```

Comment:
Explain what metered (or metering) is? At least for newcomers

Reply:
Done